docs/Custom_Operation_for_GPUs.md

```diff
@@ -558,543 +558,313 @@ The values have been computed correctly for forward operation, however, the gene
+When using custom_partitioning, it isn't needed to define the JAX config experimental_xmap_spmd_lowering and experimental_xmap_spmd_lowering_manual as was the case with the now deprecated xmap.
 
-```python
-with mesh:
-
-    pjitted = pjit(
-        partial(xmap_rms_norm, device_count=jax.local_device_count()),
-        # Shard x by batch dimension and replicate weight on all devices.
-        in_shardings=(
-            PartitionSpec("x", None, None),
-            PartitionSpec(None, None),
-        ),
-        # Shard the output by batch dimension.
-        out_shardings=PartitionSpec("x", None, None),
-    )
-    print(pjitted.lower(x, weight).compile().runtime_executable().hlo_modules()[0].to_string())
-    out = pjitted(x, weight)
-
-jnp.allclose(ref, out, atol=1e-2, rtol=1e-2)
-```
-```python
-HloModule pjit__unnamed_wrapped_function_, entry_computation_layout={(bf16[4,512,512]{2,1,0},bf16[512,512]{1,0})->bf16[4,512,512]{2,1,0}}
+We create an helper function to help with all the JAX registration needed.
 
-ENTRY %main.17_spmd (param: bf16[4,512,512], param.1: bf16[512,512]) -> bf16[4,512,512] {
-  %param = bf16[4,512,512]{2,1,0} parameter(0), sharding={devices=[8,1,1]0,1,2,3,4,5,6,7}, metadata={op_name="pjit(<unnamed wrapped function>)/jit(main)/xmap(rms_norm)/squeeze[dimensions=(0,)]" source_file="/tmp/ipykernel_25235/3123505662.py" source_line=13}
-  %param.1 = bf16[512,512]{1,0} parameter(1), sharding={replicated}, metadata={op_name="pjit(<unnamed wrapped function>)/jit(main)/xmap(rms_norm)/full_to_shard[axes=OrderedDict() mesh=Mesh(device_ids=array([0, 1, 2, 3, 4, 5, 6, 7]), axis_names=(\'x\',)) manual_axes=(\'x\',)]" source_file="/tmp/ipykernel_25235/3123505662.py" source_line=13}
-  %custom-call.0 = (bf16[4,512,512]{2,1,0}, f32[4]{0}) custom-call(bf16[4,512,512]{2,1,0} %param, bf16[512,512]{1,0} %param.1), custom_call_target="rms_forward_affine_mixed_dtype", operand_layout_constraints={bf16[4,512,512]{2,1,0}, bf16[512,512]{1,0}}, api_version=API_VERSION_STATUS_RETURNING, metadata={op_name="pjit(<unnamed wrapped function>)/jit(main)/xmap(rms_norm)/rms_norm_fwd[eps=1e-05]" source_file="/tmp/ipykernel_25235/3343076723.py" source_line=8}, backend_config="\004\000\000\000\000\000\004\000\361h\343\210\265\370\344>\000\000\000\000\000\000\000\000\000\000\000\000\027\177\000\000"
-  ROOT %get-tuple-element = bf16[4,512,512]{2,1,0} get-tuple-element((bf16[4,512,512]{2,1,0}, f32[4]{0}) %custom-call.0), index=0, metadata={op_name="pjit(<unnamed wrapped function>)/jit(main)/xmap(rms_norm)/rms_norm_fwd[eps=1e-05]" source_file="/tmp/ipykernel_25235/3343076723.py" source_line=8}
-}
-```
 ```python
-True
+def register_primitive(cls):
+    """
+    register jax primitive
+
+    The order of calls.
+
+    Inner, only the basic to wrap the TE XLA(not JAX) custom_call itself.
+    - Impl to TE XLA custom_call in C.
+    - abstract to know the static shapes
+    - lower to StableHLO TE XLA custom_call.
```

Comment:
No abreviation (Impl). Make it consistent with a capital at the start and a period at the end.